_posts/2024-10-02-debug-with-maven-wildfly-plugin.adoc

```diff
@@ -0,0 +1,198 @@
+            <configuration>
+                <provisioning-dir>${jboss.home}</provisioning-dir>
+                <galleon-options>
+                    <jboss-fork-embedded>${galleon.fork.embedded}</jboss-fork-embedded>
+                </galleon-options>
+                <feature-packs>
+                    <feature-pack>
+                        <groupId>${server.test.feature.pack.groupId}</groupId>
+                        <artifactId>${server.test.feature.pack.artifactId}</artifactId>
+                        <version>${wildfly.version}</version>
+                        <inherit-configs>false</inherit-configs>
+                        <included-configs>
+                            <config>
+                                <model>standalone</model>
+                                <name>standalone-full.xml</name>
+                            </config>
+                            <config>
+                                <model>standalone</model>
+                                <name>standalone.xml</name>
+                            </config>
+                        </included-configs>
+                        <excluded-packages>
+                            <name>docs.schema</name>
+                            <name>appclient</name>
+                            <name>domain</name>
+                        </excluded-packages>
+                    </feature-pack>
+                </feature-packs>
+                <channels>
+                    <channel>
+                        <manifest>
+                            <groupId>org.jberet</groupId>
+                            <artifactId>jberet-channel-manifest</artifactId>
+                            <version>${version.jberet}</version>
+                        </manifest>
+                    </channel>
+                </channels>
+            </configuration>
+        </execution>
+        ...
+    </executions>
+</plugin>
+----
+
+In the above configuration, the `provision` goal is configured to run in the `generate-test-resources` phase. In this execution, the `feature-packs` section is used to override the feature packs to be used by the provisioned server, and the `channels` section is using the `jberet-channel-manifest` channel to override the `batch-jberet` module to be used in the provisioned server. Please note that you need to use the `channels` with the `feature-packs` together. Here are the properties used by the configuration:
+
+[source,xml]
+----
+<properties>
+    ...
+    <version.jberet>3.0.0.Final</version.jberet>
+    ...
+    <version.maven.wildfly.plugin>5.0.1.Final</version.maven.wildfly.plugin>
+    <jboss.home>${project.build.directory}${file.separator}wildfly</jboss.home>
+    <wildfly.version>33.0.2.Final</wildfly.version>
+    <server.test.feature.pack.groupId>org.wildfly</server.test.feature.pack.groupId>
+    <server.test.feature.pack.artifactId>wildfly-ee-galleon-pack</server.test.feature.pack.artifactId>
+    ...
+</properties>
+----
+
```

Comment:
Here I think you need a section explaining what the `jberet-channel-manifest` channel is. People likely don't really know what a channel is, and they don't know that JBeret publishes one. And you need to note that the 3.0.0.Final version of that manifest specifies use of JBeret 3.0.0 while WildFly 33.0.2 used 2.2.1.

Reply:
done